inc/views/base/calendar/addevent.twig

```diff
@@ -0,0 +1,320 @@
+                            {% for calendar in calendar_select %}
+                                <option value="{{ calendar.cid }}"{% if calendar.selected %} selected="selected"{% endif %}>{{ calendar.name }}</option>
+                            {% endfor %}
+                        </select>
+                    </td>
+                </tr>
+            {% else %}
+                <input type="hidden" name="calendar" value="{{ calendar.cid }}" />
+            {% endif %}
+            <tr>
+                <td class="trow2" style="vertical-align: top;"><strong>{{ lang.event_date }}</strong></td>
+                <td class="trow1">
+                    <script type="text/javascript">
+                    <!--
+                        function toggleType()
+                        {
+                            if($("#type_single").prop("checked"))
+                            {
+                                $("#ranged_selects").hide();
+                                $("#single_selects").show();
+                            }
+                            else
+                            {
+                                $("#single_selects").hide();
+                                $("#ranged_selects").show();
+                                toggleRepeats();
+                            }
+                        }
+
+                        function toggleRepeats()
+                        {
+                            var repeats = $("#repeats").get(0).selectedIndex;
+                            $(".repeats_type").each(function()
+                            {
+                                $(this).hide();
+                            });
+                            if($("#repeats_type_"+repeats).length > 0)
+                            {
+                                $("#repeats_type_"+repeats).show();
+                            }
+                        }
+                    // -->
+                    </script>
+                    <dl style="margin-top: 0; margin-bottom: 0; width: 100%;">
+                        <dt><input type="radio" name="type" value="single"{% if select.type_single %} checked="checked"{% endif %} id="type_single" onclick="toggleType();" /> <label for="type_single"><strong>{{ lang.type_single }}</strong></label></dt>
+                        <dd style="margin-top: 4px;" id="single_selects">
+                            <select name="single_day">
+                                {% for day in days %}
+                                    <option value="{{ day.day }}"{% if day.single_day %} selected="selected"{% endif %}>{{ day.day }}</option>
+                                {% endfor %}
+                            </select>
+                            &nbsp;
+                            <select name="single_month">
+                                <option value="1"{% if select.single_month.1 %} selected="selected"{% endif %}>{{ lang.month_1 }}</option>
```

Comment:
Not a big deal, but repeated blocks like these where it's easy to forget to update a single number make me anxious. The problem with using a loop here would be the month language variable, right? If so, you an kind of cheat:

```
{% for i in 1..12 %}
    <option value="{{ i }}"{% if select.single_month[i] %} selected="selected"{% endif %}>{{ attribute(lang, 'month_' ~ i) }}</option>
{% endfor %}
```

The trick here is using the `[]` square brackets for the condition, and the [`attribute`](https://twig.symfony.com/doc/2.x/functions/attribute.html) core Twig function to get the language variable (with string concatenation using the `~` operator).

I __believe__ that should work, but I haven't tested it at all.